app/src/main/java/unipd/se18/ocrcamera/IngredientDetailsFragment.java

```diff
@@ -0,0 +1,171 @@
+package unipd.se18.ocrcamera;
+
+import android.app.SearchManager;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.v4.app.DialogFragment;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.Window;
+import android.widget.Button;
+import android.widget.TextView;
+
+import com.android.volley.Request;
+import com.android.volley.Response;
+import com.android.volley.VolleyError;
+import com.android.volley.toolbox.JsonObjectRequest;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.Iterator;
+
+
+/**
+ * Dialog Fragment that shows details about the ingredient selected such as name, description,
+ * function and an extract from wikipedia.
+ * @author Francesco Pham
+ */
+public class IngredientDetailsFragment extends DialogFragment {
+
+    private static final String TAG = "IngredDetailsFragment";
+
+    private static final String ARG_NAME = "name";
+    private static final String ARG_DESCRIPTION = "description";
+    private static final String ARG_FUNCTION = "function";
+
+    public IngredientDetailsFragment() {
+        // Required empty public constructor
+    }
+
+    /**
+     * Method for the fragment creation
+     * @param inciName Title to show in the fragment
+     * @param description Description of the ingredient
+     * @param function Functions of the ingredient
+     * @return
```

Comment:
miss return explanation

Reply:
I forgot that sorry, I will add it immediately.